client/data/plugins/use-site-plugins-query.ts

```diff
@@ -0,0 +1,50 @@
+import { useQuery, UseQueryResult } from '@tanstack/react-query';
+import wpcomRequest from 'wpcom-proxy-request';
+import type { SiteId, SiteSlug } from 'calypso/types';
+
+export type SitePlugin = {
+	active: boolean;
+	author: string;
+	author_url: string;
+	autoupdate: boolean;
+	description: string;
+	display_name: string;
+	name: string;
+	network: boolean;
+	plugin_url: string;
+	slug: string;
+	uninstallable: boolean;
+	version: string;
+};
+
+export type SitePluginsResponse = {
+	file_mod_capabilities: {
+		modify_files: boolean;
+		autoupdate_files: boolean;
+	};
+	plugins: SitePlugin[];
+};
+
+/**
+ * Fetches the plugins for a given site
+ * @param siteIdOrSlug The site ID or slug
+ */
+export const useSitePluginsQuery = (
```

Comment:
Can we re-use [SitePlugin](https://github.com/Automattic/wp-calypso/blob/04ab2d1b8f4b657920efde3fd334f404500b882f/packages/wpcom.js/src/lib/site.plugin.js#L27) for this?

Reply:
Probably yes, I'll take a closer look.